src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3897,11 +3897,10 @@ GenTree* Lowering::LowerHWIntrinsicTernaryLogic(GenTreeHWIntrinsic* node)
                     case TernaryLogicUseFlags::AB:
                     {
                         // Swap the operands here to make the containment checks in codegen significantly simpler
-                        std::swap(node->Op(2), node->Op(3));
-                        std::swap(node->Op(1), node->Op(2));
+                        std::swap(node->Op(1), node->Op(3));
 
                         // Make sure we also fixup the control byte
-                        control = TernaryLogicInfo::GetTernaryControlByte(info, C, A, B);
+                        control = TernaryLogicInfo::GetTernaryControlByte(info, C, B, A);
```

Comment:
Not understanding why this fixes the problem; I think it might instead introduce a different one...

We start with `ABC` where we're using `AB` and `C` is unused. (A == op1, B == op2, C == op3)

We then swap `op2/op3` giving us `ACB` (A == op1, C == op2, B == op3)
We then swap `op1/op2` giving us `CAB` (C == op1, A == op2, B == op3)

This should change the use to `BC` leaving `A`  (op1) as unused.

-----------------

All you've done here is effectively change it from `ABC` to `CBA` (C == op1, B == op2, A == op3)

This will mean that the operand order has changed for used operands, which can negatively impact certain operations.

I think rather the issue is that `GetTernaryControlByte` should've been `(info, B, C, A)` instead. Because it was `AB` with C unused and we want `BC` with `A` unused

Reply:
Got it. Yes, I didn't understand these semantics, so that was my best (and wrong) guess. Fixed it as per your suggestion.